ui/src/pages/AllStories.tsx

```diff
@@ -0,0 +1,250 @@
+import AppBar from "@material-ui/core/AppBar";
+import Checkbox from "@material-ui/core/Checkbox";
+import Switch from "@material-ui/core/Switch";
+import Tab from "@material-ui/core/Tab";
+import Tabs from "@material-ui/core/Tabs";
+import { useState } from "react";
+import * as React from "react";
+import styled from "styled-components";
+import useSWR from "swr";
+
+import { a11yProps, AllStoriesTabs } from "../components/AllStoriesTabs";
+import VirtualizedTable from "../components/VirtualizedTable";
+import { Story } from "../types/index";
+
+function createData(
+  id: number,
+  title: string,
+  current_city: string,
+  year: number,
+  is_visible: boolean,
+  author_first_name: string,
+  author_last_name: string,
+  author_country: string
+) {
+  const author_name = `${author_first_name} ${author_last_name}`;
+  return {
+    id,
+    title,
+    current_city,
+    year,
+    author_name,
+    author_country,
+    is_visible,
+  };
+}
+
+const StyledAppBar = styled(AppBar)`
+  //styling
+`;
+
+export const AllStories: React.FC = () => {
+  const { data: allStories, error } = useSWR<Story[]>("/api/stories");
+  let rows = [];
+
+  // if (allStories===undefined) return;
+  if (allStories) {
+    rows = allStories.map((story, i) =>
+      createData(
+        i,
+        story.title,
+        story.current_city,
+        story.year,
+        story.is_visible,
+        story.author_first_name,
+        story.author_last_name,
+        story.author_country
+      )
+    );
+  }
+
+  const [tabValue, setTabValue] = React.useState(0);
+  const [visibleState, setVisibleState] = React.useState([]);
+  const [tableData, setTableData] = useState(rows);
+  const [selectedRowIds, setSelectedRowIds] = useState([]);
+  const [order, setOrder] = useState("desc");
+  const [orderBy, setOrderBy] = useState("id");
+
+  const handleChange = (e, d) => {
+    if (e.target.checked) {
+      setVisibleState((prevState) => [...prevState, d.id]);
+    } else {
+      setVisibleState((prevState) => prevState.filter((e) => e !== d.id));
+    }
+  };
+
+  const handleCheckedAll = () => {
+    setSelectedRowIds((prevState) =>
+      prevState.length === rows.length ? [] : rows.map((d) => d.id)
```

Comment:
I'm not too familiar with reach but where is `d` defined?

Reply:
d is defined in the function (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions)